### DISCUSSION TITLE
中文文章段落内容支持缩进

### DISCUSSION THREAD

**Discussion by iambinlin** [YunYouJun/hexo-theme-starry#282] [Ideas 奇思妙想]
## 文章内容段落支持缩进

当然也可以通过修改`styl`文件来做。

```styl
html.sr {
  .post-content {
    &>p {
      text-indent 2em;
    }
  }
}

```

效果截图：

<img width="1395" alt="image" src="https://user-images.githubusercontent.com/39019913/163422743-59c088fe-025f-479c-8af8-8ac7a40457ea.png">

emmm，只是讨论下是否有默认支持的必要性～

**Comment by YunYouJun** [comment]
一些使用 Markdown 编写的技术文章会采取顶格排版的方式。
我觉得这可以成为通过开关开启的功能，但不是默认的样式。

您有兴趣创建 PR 吗？

**Reply by iambinlin** [reply]
可以的，我试试，我之前想了下，可以通过给每一篇 `post`的文章配置上加个配置参数，好区分是`技术文章`还是`生活随笔`。

**Reply by YunYouJun** [reply]
对，我觉得作为 front-matter 是最合适的。
名称我建议是：

```md
---
indent: true
---
```

如果有别的想法可以提。

**Reply by iambinlin** [reply]
> 对，我觉得作为 front-matter 是最合适的。 名称我建议是：
> 
> ```lisp
> ---
> indent: true
> ---
> ```
> 
> 如果有别的想法可以提。

可以的，跟你碰一块儿去了，你觉得这个 `indent`要不要加个联合类型`boolean` | `number`。默认 `1em`，可以让他自己控制

**Reply by YunYouJun** [reply]
🉑️

---

此外是否考虑看下 [valaxy](https://github.com/YunYouJun/valaxy)，我使用 TS 重新设计的一个静态博客框架，目的是解决 Hexo 现存的热更新、缺乏类型提示等问题。
目前支持了默认的 valaxy-theme-yun，我打算在它足够稳定后，将博客迁移至该处。
此外正如我们提到的 front-matter 各种配置这类功能，我希望可以将博客常用的功能都在 Valaxy 中集成并默认提供。
譬如 `indent` 作为一个默认支持的配置项，使用 Valaxy 编写的任意其他主题都可以共用该功能，而无需主题自己实现。

**Reply by iambinlin** [reply]
好嘞，我先熟悉熟悉`pug`和`hexo`。😆
